docs/modules/data-structures/pages/vector-collections.adoc

```diff
@@ -0,0 +1,472 @@
+
+[tabs]
+====
+Java::
++
+--
+[source,java]
+----
+VectorCollection vectorCollection = VectorCollection.getCollection(hazelcastInstance, vectorCollectionConfig);
+----
+--
+====
+
+* Method `getCollection(HazelcastInstance instance, VectorCollectionConfig collectionConfig)`.
+** If a collection with the provided name does not exist, the system will create the collection with the configuration created implicitly during static or dynamic configuration of the cluster. If the configuration does not exist, an error will be thrown.
+** If a collection with the provided name exists, it will be returned.
+
+[tabs]
+====
+Java::
++
+--
+[source,java]
+----
+VectorCollection vectorCollection = VectorCollection.getCollection(hazelcastInstance, collectionName);
+----
+--
+====
+
+=== Optimize collection
+
+An optimization operation may be needed in the following cases:
+
+* To permanently delete nodes from the graph that were marked for removal.
+* After adding a significant number of vectors.
+* If some nodes become unreachable and the search results return fewer vectors than expected.
+
+NOTE: The optimization operation is a time-consuming process, and no mutating operations are allowed during this process.
+
+[tabs]
+====
+Java::
++
+--
+[source,java]
+----
+VectorCollection vectorCollection = VectorCollection.getCollection(hazelcastInstance, collectionName);
+CompletionStage<Void> result = vectorCollection.optimizeAsync("index_name");
+----
+--
+====
+
+== Manage data
+All methods of `VectorCollection` that work with collection data are asynchronous. The result is returned as a `CompletionStage`. A collection interacts with entries in the form of documents (`VectorDocument`). Each document comprises a value and one or more vectors associated with that value.
+
+=== Create document
+To create a document use statics the methods of `VectorDocument` and `VectorValues` classes.
+
+Example single index vector collection:
+[tabs]
+====
+Java::
++
+--
+[source,java]
+----
+VectorDocument<String> document = VectorDocument.of(
+        "document value",
+        VectorValues.of(
+                new float[]{0.2f, 0.9f, -1.2f}
+        )
+);
+----
+--
+====
+
+For multi-index collections, specify the names of the indexes to which the vectors belong:
+[tabs]
+====
+Java::
++
+--
+[source,java]
+----
+VectorDocument<String> document = VectorDocument.of(
+        "some document value",
+        VectorValues.of(
+                Map.of(
+                        "index1", new float[] {1.2f, -0.3f, 2.2f},
+                        "index2", new float[] {0.2f, 1.3f, -0.2f}
+                )
+        )
+);
+----
+--
+====
+
+
+=== Put entries
+To put single document to a vector collection, use the method `putAsync`, `putIfAbsent` or `setAsync` of the `VectorCollection` class.
+[tabs]
+====
+Java::
++
+--
+[source,java]
+----
+VectorCollection<String, String> vectorCollection = VectorCollection.getCollection(hazelcastInstance, collectionName);
```

Comment:
if you use one example throughout entire document, you will not have to repeat collection creation
see https://github.com/hazelcast/hz-docs/pull/1124/files#r1632795948

Reply:
7e1b5c3452fd6c95ac9cd6d02014415617138c44